core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java

```diff
@@ -314,6 +315,34 @@ public static String getClientHostName() {
     return getLocalHostName();
   }
 
+  /**
+   * Gets a local node name from configuration if it is available, falling back on localhost lookup.
+   *
+   * @return the local node name
+   */
+  public static synchronized String getLocalNodeName() {
```

Comment:
why is this `synchronized`? And, can this replace `getClientHostName`?

Reply:
removed `synchronized`. `getClientHostName` is only for clients, while this method works for all process types.